staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -1365,9 +1367,26 @@ func (e *Store) CompleteWithOptions(options *generic.StoreOptions) error {
 		)
 	}
 
+	if opts.CountMetricPollPeriod > 0 {
+		e.startObservingCount(e.DefaultQualifiedResource.String(), opts.StorageConfig.Prefix+prefix, opts.CountMetricPollPeriod)
```

Comment:
don't run forever, make sure we have a stop channel and non-nil Destroy() func that will stop this

Reply:
for example:
```go
stopMetricsCh := make(chan struct{})
destroy := e.DestroyFunc
e.DestroyFunc = func(){
  close(stopMetricsCh)
  if destroy != nil {
    destroy()
  }
}

e.startObservingCount(..., stopMetricsCh)
```

Reply:
don't join `opts.StorageConfig.Prefix+prefix` here... the storage is responsible for dealing with prepending `opts.StorageConfig.Prefix`

also, don't assume `prefix` is the key... call `e.KeyRootFunc()` to get the key

Reply:
I started to return a destroy function from 'start...', which is functionally equivalent.